frame/staking/src/lib.rs

```diff
@@ -483,7 +489,11 @@ impl<AccountId, Balance: HasCompact + Copy + Saturating + AtLeast32BitUnsigned +
 					false
 				}
 			})
-			.collect();
+			.collect::<Vec<_>>()
+			.try_into()
```

Comment:
```suggestion
			.try_collect()
```

Reply:
(See `TryCollect` in `frame-support`)

Reply:
Looking at this closer, `try_collect` is not (and should not) be implemented for a type that does not implement `ExactSizeIterator`, which gives us the `len` fn. 

If we do a blanket impl just for `Iterator`, we would have to use `size_hint`, which gives a general upper bound, but according to the docs  on the `Filter` iter does not guarantee that the upper bound will be respected: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.size_hint . Specifically it says "...  but must not be trusted to e.g., omit bounds checks in unsafe code."

However there might be a way to this where we only implement it for a specific subset of Filter<I, P> where I is bound by `ExactSizeIterator`, however I have not been able to figure out how to do this. Specifically I keep getting an error along the lines of 

```
  | |_^ conflicting implementation for `std::iter::Filter<_, _>`
    |
    = note: upstream crates may add a new impl of trait `std::iter::ExactSizeIterator` for type `std::iter::Filter<_, _>` in future versions
    ```

Reply:
Right, `Filter` cannot work with `try_collect`, my bad. 

I also had some thoughts about why of it here: https://github.com/paritytech/substrate/pull/10590#issue-1094323971